python/pyiceberg/table/__init__.py

```diff
@@ -341,7 +346,18 @@ def plan_files(self) -> Iterator[FileScanTask]:
             all_files = files(io.new_input(manifest.manifest_path))
             matching_partition_files = filter(partition_filter, all_files)
 
-            yield from (FileScanTask(file) for file in matching_partition_files)
+            def _check_content(file: ManifestFile) -> ManifestFile:
+                try:
+                    if file.content == ManifestContent.DELETES:
+                        raise ValueError("PyIceberg does not support deletes: https://github.com/apache/iceberg/issues/6568")
```

Comment:
shall we have a test to cover this ?

Reply:
Good call, I ran some table scans locally, but great idea to have this unit-tested as well 👍🏻